src/analysis/special_functions/pow.lean

```diff
@@ -1168,11 +1159,34 @@ begin
           nnreal.rpow_le_rpow_of_exponent_ge (bot_lt_iff_ne_bot.mpr h) hx1 hyz] }
 end
 
-lemma rpow_le_one {x : ennreal} {z : ℝ} (hx2 : x ≤ 1) (hz : 0 ≤ z) : x^z ≤ 1 :=
+lemma rpow_lt_one {x : ennreal} {z : ℝ} (hx : x < 1) (hz : 0 < z) : x^z < 1 :=
 begin
-  lift x to ℝ≥0 using ne_of_lt (lt_of_le_of_lt hx2 coe_lt_top),
-  simp at hx2,
-  simp [coe_rpow_of_nonneg _ hz, nnreal.rpow_le_one hx2 hz]
+  lift x to ℝ≥0 using ne_of_lt (lt_of_lt_of_le hx le_top),
+  simp at hx,
```

Comment:
I know some cases of this were already in the library, but this kind of nonterminal use of `simp` should be avoided. The reason is that when `hx` is used in the next line, it's expected to be in a certain shape, as an argument to `rpow_lt_one`. If the simp set changes in the future, it could be simplified differently and would no longer fit there. This kind of breakage can be very hard to debug.

The easy fix is to change these nonterminal `simp` calls to `simp only`, using `squeeze_simp`.